src/Simplex/Messaging/Agent/NtfSubSupervisor.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Simplex.Messaging.Agent.NtfSubSupervisor
+  ( NtfSubSupervisor (..),
+    RcvQueueNtfCommand (..),
+    newNtfSubSupervisor,
+    addNtfSubWorker,
+    addNtfSubSMPWorker,
+    removeNtfSubWorker,
+    removeNtfSubSMPWorker,
+    nsUpdateToken,
+    nsUpdateToken',
+    nsRemoveNtfToken,
+    addRcvQueueToNtfSubQueue,
+    addRcvQueueToNtfSubQueue',
+    closeNtfSubSupervisor,
+  )
+where
+
+import Control.Concurrent.Async (Async, uninterruptibleCancel)
+import Control.Monad
+import Control.Monad.IO.Unlift (MonadUnliftIO)
+import Simplex.Messaging.Agent.Env.SQLite
+import Simplex.Messaging.Agent.Monad (AgentMonad)
+import Simplex.Messaging.Agent.Store
+import Simplex.Messaging.Client.Agent ()
+import Simplex.Messaging.Notifications.Client
+import Simplex.Messaging.Notifications.Protocol (NtfTknStatus (NTActive))
+import Simplex.Messaging.Protocol (ProtocolServer (..), SMPServer)
+import Simplex.Messaging.TMap (TMap)
+import qualified Simplex.Messaging.TMap as TM
+import UnliftIO (async)
+import UnliftIO.STM
+
+data NtfSubSupervisor = NtfSubSupervisor
+  { ntfTkn :: TVar (Maybe NtfToken),
+    ntfSubQ :: TBQueue (RcvQueue, RcvQueueNtfCommand),
+    ntfSubWorkers :: TMap NtfServer (TMVar (), Async ()),
+    ntfSubSMPWorkers :: TMap SMPServer (TMVar (), Async ())
+  }
+
+data RcvQueueNtfCommand = RQNCCreate | RQNCDelete
```

Comment:
I thought this type would include all queue actions in two constructor, possibly with actions for SMP and NTF servers in two different constructors.

What part would be adding these actions to the DB? How would this part know which actions to add?